ext/xml/tests/XML_OPTION_PARSE_HUGE.phpt

```diff
@@ -0,0 +1,88 @@
+--TEST--
+Request #68325 (XML_OPTION_PARSE_HUGE cannot be set for xml_parser_create)
+--EXTENSIONS--
+xml
+--FILE--
+<?php
+
+function logName(string $str) {
+    if (strlen($str) > 20) {
+        echo substr($str, 0, 20) . "...\n";
+    } else {
+        echo $str . "\n";
+    }
+}
+
+function createParser(bool $huge) {
+    $parser = xml_parser_create();
+    echo "old option value: "; var_dump(xml_parser_get_option($parser, XML_OPTION_PARSE_HUGE));
+    xml_parser_set_option($parser, XML_OPTION_PARSE_HUGE, $huge);
+    echo "new option value: "; var_dump(xml_parser_get_option($parser, XML_OPTION_PARSE_HUGE));
+    xml_set_element_handler($parser, function($parser, $data) {
+        echo "open: ";
+        logName($data);
+    }, function($parser, $data) {
+    });
+    return $parser;
+}
+
+// Construct XML that is too large to parse without XML_OPTION_PARSE_HUGE
+$long_text = str_repeat("A", 1000 * 1000 * 5 /* 5 MiB */);
```

Comment:
MiB is 1024 * 1024, 1000 * 1000 is MB.

Reply:
Good catch, must've been sleepy when I wrote this :-)